src/output-tx.c

```diff
@@ -71,20 +74,60 @@ int OutputRegisterTxLogger(const char *name, AppProto alproto, TxLogger LogFunc,
     op->output_ctx = output_ctx;
     op->name = name;
     op->module_id = (TmmId) module_id;
+    op->tc_log_progress = 0;
+    op->ts_log_progress = 0;
 
-    if (list == NULL)
+    if (list == NULL) {
+        op->id = 1;
         list = op;
-    else {
+    } else {
         OutputTxLogger *t = list;
         while (t->next)
             t = t->next;
+        op->id = t->id * 2;
         t->next = op;
     }
 
     SCLogDebug("OutputRegisterTxLogger happy");
     return 0;
 }
 
+int OutputRegisterTxLoggerWithProgress(const char *name, AppProto alproto,
```

Comment:
instead of duplicating OutputRegisterTxLogger I think it's cleaner to make OutputRegisterTxLogger a wrapper to this function with preset values for int tc_log_progress, int ts_log_progress. I think it's possible to simply set it to the result AppLayerParserGetStateProgressCompletionStatus.